src/argparse.zig

```diff
@@ -0,0 +1,297 @@
+const std = @import("std");
+
+const main = @import("main");
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+const ListUnmanaged = main.ListUnmanaged;
+
+pub fn Parser(comptime T: type, comptime callback: ?fn(self: T) anyerror!void) type {
+	return struct {
+		const Self = @This();
+		pub const Args = T;
+
+		pub fn parse(allocator: NeverFailingAllocator, args: []const u8) !ParseResult(T) {
+			const result = _parse(allocator, args);
+			if(callback != null and result == .success) {
+				try callback.?(result.success);
+			}
+			return result;
+		}
+		pub fn _parse(allocator: NeverFailingAllocator, args: []const u8) ParseResult(T) {
+			switch(@typeInfo(T)) {
+				inline .@"struct" => |s| {
+					return parseStruct(s, allocator, args);
+				},
+				inline .@"union" => |u| {
+					if(u.tag_type == null) @compileError("Union must have a tag type");
+					return parseUnion(u, allocator, args);
+				},
+				else => @compileError("Only structs and unions are supported"),
+			}
+		}
+		fn parseStruct(comptime s: std.builtin.Type.Struct, allocator: NeverFailingAllocator, args: []const u8) ParseResult(T) {
+			var result: T = undefined;
+
+			var split = std.mem.splitScalar(u8, args, ' ');
+			var offset: usize = 0;
+			var count: usize = 0;
+
+			inline for(s.fields) |field| {
+				if(!std.mem.eql(u8, field.name, "flags")) {
```

Comment:
How do flags work exactly?
Also this check should be moved to the PR that actually adds flag support.

Reply:
They don't, it was an idea I never got to implement, I'll remove it.